### PR TITLE
Generate a string as the random suffix for the temporary table name when replacing tables

### LINKED ISSUE #70319: Test 03237_create_or_replace_view_atomically_with_atomic_engine is flaky
https://play.clickhouse.com/play?user=play#c2VsZWN0IAp0b1N0YXJ0T2ZIb3VyKGNoZWNrX3N0YXJ0X3RpbWUpIGFzIGQsCmNvdW50KCksIGdyb3VwVW5pcUFycmF5KHB1bGxfcmVxdWVzdF9udW1iZXIpLCAgYW55KHJlcG9ydF91cmwpCmZyb20gY2hlY2tzIHdoZXJlIHRvZGF5KCkgLSBJTlRFUlZBTCAnMyBtb250aHMnIDw9IGNoZWNrX3N0YXJ0X3RpbWUgYW5kIHRlc3RfbmFtZSBsaWtlICclMDMyMzdfY3JlYXRlX29yX3JlcGxhY2Vfdmlld19hdG9taWNhbGx5X3dpdGhfYXRvbWljX2VuZ2luZSUnIGFuZCB0ZXN0X3N0YXR1cyBpbiAoJ0ZBSUwnLCAnRkxBS1knKSBncm91cCBieSBkIG9yZGVyIGJ5IGQgZGVzYw==

cc @tuanpach

### PR DESCRIPTION
**Pull request by tuanpach**
### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
There is [a failed CI job](https://s3.amazonaws.com/clickhouse-test-reports/69778/2d81c38874958bd9d54a25524173bdb1ddf2b75c/stateless_tests__release_.html) which is triggered by [03237_create_or_replace_view_atomically_with_atomic_engine](https://github.com/ClickHouse/ClickHouse/blob/master/tests/queries/0_stateless/03237_create_or_replace_view_atomically_with_atomic_engine.sh).

The temporary table name `_tmp_replace_1e7a789f928286ed_f66a` were duplicated. There were different UUIDs with the same table name: `_tmp_replace_1e7a789f928286ed_f66a.f1e1d4f4-6b6d-4d0a-befb-5f7b22b60604` and `_tmp_replace_1e7a789f928286ed_f66a.572c1488-1097-4086-a70e-4f2c1d78a03a`.

This PR is to generate a string as the random suffix for the temporary table name when replacing tables.
Closes #70319

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit c2b83708e2621b3db8c80a78d7e8b4ebe262e907 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70330/c2b83708e2621b3db8c80a78d7e8b4ebe262e907/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/62cdedd57e12d762cf11a858e561eba342d729dc/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Interpreters/InterpreterCreateQuery.cpp [hunk 1]
```diff
@@ -2333,6 +2337,18 @@ void InterpreterCreateQuery::processSQLSecurityOption(ContextPtr context_, ASTSQ
         context_->checkAccess(AccessType::ALLOW_SQL_SECURITY_NONE);
 }
 
+String InterpreterCreateQuery::getRandomSuffix(size_t size)
```
Comment (pufit) [inline src/Interpreters/InterpreterCreateQuery.cpp:2340]: Probably we can reuse this function
https://github.com/ClickHouse/ClickHouse/blob/master/src/Common/getRandomASCIIString.cpp#L8

**Comment by pufit** [comment]
Can we add a test for this somehow?

**Comment by tuanpach** [comment]
@pufit  Do you have another comment on this PR?

**Comment by tuanpach** [comment]
> (Note for future PRs: let's wait for the assignee to hit the 'approve' button first before merging, thanks)

Noted. The assignee is on leave, and this PR is quite simple. So I thought I could merge it to fix the CI.
Nevertheless, I will wait for it to be approved before merging next time.
Thanks.

**Comment by tavplubix** [comment]
Code review is not mandatory for small/simple PRs

**Pull request merged**